model/DataStore/MetaDataDeliverySyncTask.php

```diff
@@ -108,25 +131,22 @@ private function getPersistDataService(): PersistDataService
         return $this->getServiceLocator()->get(PersistDataService::class);
     }
 
-    /**
-     * @throws common_exception_Error
-     * @throws core_kernel_persistence_Exception
-     */
-    private function prepareData($params)
+    private function prepareMetaData($params)
     {
-        $params['count'] = $params['count'] ?? 0;
-        if (!isset($params['deliveryMetaData'], $params['testMetaData'], $params['testUri'], $params['itemMetaData'])) {
-            $compiler = $this->getMetaDataCompiler();
+        $compiler = $this->getMetaDataCompiler();
+        if ($params[self::INCLUDE_DELIVERY_METADATA_PARAM_NAME]) {
```

Comment:
For me it is surprising that same parameter can have both delivery or test id, and that test will be treated as delivery. I think this way it could be confusing, better not mix it up. If it is a test triggering this task, then it is compiled 2 times?

Reply:
Fixed